src/resmom/mom_comm.c

```diff
@@ -278,8 +317,43 @@ task_save(pbs_task *ptask)
 			return (-1);
 		}
 	}
+	/* save all obits */
+
+	/* save current position so we can update the obit count later */
+	off_t savedpos = lseek(fds, (off_t)0, SEEK_CUR);
+	if (savedpos < 0)
+		goto failedwrite;
+
+	unsigned int obitcount = 0;
+	if (ensure_write(fds, (char *)&obitcount, sizeof(unsigned int)) != 0)
+		goto failedwrite;
+
+	obitent *pobit = (obitent *)GET_NEXT(ptask->ti_obits);
+	while (pobit) {
+		obitcount++;
+		if (ensure_write(fds, (char *)&pobit->oe_type, sizeof(pobit->oe_type)) != 0)
+			goto failedwrite;
+		if (ensure_write(fds, (char *)&pobit->oe_u.oe_tm.oe_fd, sizeof(pobit->oe_u.oe_tm.oe_fd)) != 0)
```

Comment:
@bayucan Please correct me if I am wrong...
but @zv0n I don't think saving obits help us regain connection with clients waiting for obits as if mom process goes then oe_fd will be released and clients (aka pbsdsh or pbs_tmrsh) connected to oe_fd will get disconnect error right? like below:
```
pbsroot@node1:~$ qsub -I
qsub: waiting for job 2.node1 to start
qsub: job 2.node1 ready

pbsroot@node1:~$ pbsdsh sleep 100 
pbsdsh: Event poll failed, error not connected (17002)
```
I just killed pbs_mom when sleep 100 process started

Reply:
Yes, that's one of the issues I've written about in the initial pull request comment, pbsdsh fails when mother superior is restarted. But if you restart one of the sisters, pbsdsh continues going and expects obit from the sisters, so if we don't save the obit information, it would result in an infinite loop within pbsdsh.

In this commit I wanted to create server-side (mom) support for the multi-node resume and thought client side might be done in a different commit.

This problem is within pbsdsh only and I think it should be fixable, for example mpirun (at least during my testing) didn't error out during MS restart and managed to reconnect and after receiving obits terminate successfuly.

Reply:
So if I understand correctly @zv0n, then if we fix clients not to error out and wait for MS or sister nodes to come up and reconnect, then also we don't need save obit info, because once clients reconnects it will do tm_obit (to register obit) and tm_poll (wait for obit to finish) again right?

Reply:
@hirenvadalia Yes, that sounds right. If the clients can recognize when a node reconnects, so they can delete the old event and register a new obit, that would remove the need to save obits.

Reply:
exactly @zv0n!, only thing is clients has to try (may be forever or for some timeout) reconnecting to node... and once connection re-establish, no need to delete old event, just register obit for given task and wait for its finish

Reply:
Also doing above will eliminate all possible problems which you mentioned in PR description...

Reply:
Wouldn't the old event need to be deleted? `tm_obit` calls `add_event` inside the function body, which allocates a new event, wouldn't that be a problem? If we don't want to delete the old event I think we might need to add a new function that only re-sends the obit number without creating a new event.
I've only skimmed through the function, so I might be wrong about how it works though.

Reply:
Since obits are not saved in .TK file so on mom restart mom wouldn't have that information, if we save obits in .TK and recover that on mom restart, then only we need to delete that, but as we don't need to save obits so no need of deleting it... HTH!

Reply:
I see! Great, thanks!

Reply:
So, @hirenvadalia, do you think I should remove the obit-saving parts and leave this PR as is? Because so far I have no idea how to make pbsdsh operate without an open socket to MS 😅

Reply:
@zv0n, sorry for delay, I think we can proceed with this PR after removing obit-saving parts but without fixing pbsdsh/pbs_tmrsh, will not get full resume of multinode.
Based on your one of above comment, where you say mpirun was able to reconnect on MS restart and continue, so if we don't fix pbsdsh and pbs_tmrsh then only multinode jobs which uses mpirun part will able resume i think.

Reply:
@hirenvadalia I've removed the obit saving bits and did a test run, as you expected mpirun works without a problem, pbsdsh gets stuck on waiting for obit.

Reply:
Thanks @zv0n... will continue code and design review...